session-1-jigsaw-intro/07_patch_module_option/src/java.base/java/util/concurrent/ConcurrentHashMap.java

```diff
@@ -2,44 +2,68 @@
 
 import java.io.Serializable;
 import java.util.AbstractMap;
+import java.util.HashMap;
 import java.util.Map;
 import java.util.Set;
 
 public class ConcurrentHashMap<K,V> extends AbstractMap<K,V> implements ConcurrentMap<K,V>, Serializable {
-
-	public static void main(String[] args) {
-		System.out.printf("ConcurrentHashMap");
-	}
+	private final Map<K, V> innerMap;
 
 	public ConcurrentHashMap() {
-	}
-
-	public ConcurrentHashMap(Map<? extends K, ? extends V> m) {
+		this.innerMap = new HashMap<>();
 	}
 
 	public ConcurrentHashMap(int initialCapacity) {
+		this.innerMap = new HashMap<>(initialCapacity);
 	}
 
-	public V put(K key, V value) {
-		throw new RuntimeException("Not implemented <ConcurrentHashMap>");
+	public ConcurrentHashMap(int initialCapacity,
+							 float loadFactor, int concurrencyLevel) {
+		this.innerMap = new HashMap<>(initialCapacity, loadFactor);
```

Comment:
Always storing a fixed value will be difficult, since we're working with generics here and can't really change the interface visibly - at least, not without using reflection which would divert the reader from the actual topic. I tried wrapping the inner objects, but that leads to all kind of problems because ConcurrentHashMap is used internally in so many places.

I have however found an interesting and simple example. The PR for that is coming right up.